app_radio_module/src/transmitter.c

```diff
@@ -81,6 +90,36 @@ static void lora_tx_task(void *, void *, void *) {
     }
 }
 
+typedef struct {
+    double latitude;
+    double longitude;
+    double altitude;
+} gnss_data_simple;
+
+static void gnss_data_cb(const struct device *dev, const struct gnss_data *data) {
+    if (!ready_to_tx) {
+        return; // timer hasnt expired yet
+    }
+    l_lora_packet_t packet = {0};
+    int port_num = RADIO_MODULE_GNSS_DATA_PORT + RADIO_MODULE_BASE_PORT;
+    packet.port = port_num;
+    packet.payload_len = sizeof(gnss_data_simple);
+
+    gnss_data_simple gnss_data = {0};
+    gnss_data.latitude = data->nav_data.latitude / L_GNSS_LATITUDE_DIVISION_FACTOR;
+    gnss_data.longitude = data->nav_data.longitude / L_GNSS_LONGITUDE_DIVISION_FACTOR;
+    gnss_data.altitude = data->nav_data.altitude / L_GNSS_ALTITUDE_DIVISION_FACTOR;
+
+    memcpy(packet.payload, &gnss_data, sizeof(gnss_data_simple));
```

Comment:
Nothing wrong with this, but trying to optimize this code to run as fast as possible in a callback, I would just cast the uint8_t buffer to a gnss data struct and then fill the data from there.

Reply:
This isn't possible from what I can tell, and filling in the bytes of the packet by bitshifting the data is too much work.